Mathlib/Analysis/SumIntegralComparisons.lean

```diff
@@ -31,16 +30,49 @@ with a `Filter.Tendsto` to estimate some errors.
   values at integer steps aligning with the right-hand side of the interval
 * `MonotoneOn.sum_le_integral`: The sum of a monotone function along integer steps aligning with
   the left-hand side of the interval is at most the integral of the function along that interval
+* `sum_mul_Ico_le_integral_of_monotone_antitone`: the sum of `f i * g i` on an interval is bounded
+  by the integral of `f x * g (x - 1)` if `f` is monotone and `g` is antitone.
+* `integral_le_sum_mul_Ico_of_antitone_monotone`: the sum of `f i * g i` on an interval is bounded
+  below by the integral of `f x * g (x - 1)` if `f` is antitone and `g` is monotone.
 
 ## Tags
 
 analysis, comparison, asymptotics
 -/
 
 
-open Set MeasureTheory.MeasureSpace
+open Set MeasureTheory MeasureSpace
 
-variable {x₀ : ℝ} {a b : ℕ} {f : ℝ → ℝ}
+variable {x₀ : ℝ} {a b : ℕ} {f g : ℝ → ℝ}
```

Comment:
Is there a reason `a` and `b` are naturals for your results, instead of integers?

Reply:
It's just that most of the time sums are indexed by naturals. Results with integers would also make sense, but I think they would be used less often. For instance, in Carleson I need the Nat-indexed version.

Reply:
Maybe it's useful to first do it for `Int` and then specialize to `Nat`. I'll let you decide.

bors d+

Reply:
Given that a very large majority of the time sums are Nat-indexed, I'll keep the current version and leave the generalization to whoever needs it.